benchmarks/newton_solver_benchmark_set/spiegelman_et_al_2016/drucker_prager_compositions.cc

```diff
@@ -0,0 +1,711 @@
+         * equation as $\nabla \cdot (\rho \mathbf u)=0$ (compressible Stokes)
+         * or as $\nabla \cdot \mathbf{u}=0$ (incompressible Stokes).
+        *
+        * This material model is incompressible.
+         */
+        virtual bool is_compressible () const;
+
+        virtual double reference_viscosity () const;
+
+        virtual double reference_density () const;
+
+        static
+        void
+        declare_parameters (ParameterHandler &prm);
+
+        virtual
+        void
+        parse_parameters (ParameterHandler &prm);
+
+      private:
+
+        double reference_T;
+        std::vector<double> reference_T_list;
+
+        /**
+         * Defining a minimum strain rate stabilizes the viscosity calculation,
+         * which involves a division by the strain rate. Units: $1/s$.
+         */
+        bool use_deviator_of_strain_rate;
+        std::vector<double> min_strain_rate;
+        std::vector<double> min_visc;
+        std::vector<double> max_visc;
+        std::vector<double> veff_coefficient;
+        double ref_visc;
+        double reference_compressibility;
+        std::vector<double> ref_visc_list;
+
+        std::vector<double> thermal_diffusivity;
+        std::vector<double> heat_capacity;
+
+
+        std::vector<double> densities;
+        std::vector<double> thermal_expansivities;
+
+        std::vector<double> cohesion;
+        std::vector<double> phi;
+
+        std::vector<double> prefactor;
+        std::vector<double> stress_exponent;
+
+        unsigned int n_fields;
+        // averaging parameters
+        double viscosity_averaging_p;
+
+        bool use_analytical_derivative;
+
+        //optimize variables
+        const double sqrt_3 = std::sqrt(3);
+        const double sqrt_half = std::sqrt(0.5);
+        std::vector<double> sin_phi;
+        std::vector<double> cos_phi;
+
+
+
+    };
+
+  }
+}
+
+#endif
+
+#include <aspect/utilities.h>
+#include <aspect/parameters.h>
+
+using namespace dealii;
+
+namespace aspect
+{
+  namespace
+  {
+    std::vector<double>
+    get_vector_double (const std::string &parameter, const unsigned int n_fields, ParameterHandler &prm)
+    {
+      std::vector<double> parameter_list;
+      parameter_list = Utilities::string_to_double(Utilities::split_string_list(prm.get (parameter)));
+      if (parameter_list.size() == 1)
+        parameter_list.resize(n_fields, parameter_list[0]);
+
+      AssertThrow(parameter_list.size() == n_fields,
+                  ExcMessage("Length of " + parameter + " list (size "+ std::to_string(parameter_list.size()) +") must be either one,"
+                             " or n_compositional_fields+1 (= " + std::to_string(n_fields) + ")."));
+
+      return parameter_list;
+    }
+  }
+
+  namespace MaterialModel
+  {
+    template <int dim>
+    std::vector<double>
+    DruckerPragerCompositions<dim>::
+    compute_volume_fractions( const std::vector<double> &compositional_fields) const
```

Comment:
can this be replaced with `MaterialUtilities::compute_volume_fractions`?

Reply:
hmm, the implementation is slighty different, but I think the result of the function should still be exactly the same. So I will change it to use that function.